django_auth_ldap/backend.py

```diff
@@ -820,9 +820,34 @@ def _get_groups(self):
     def _bind(self):
         """
         Binds to the LDAP server with AUTH_LDAP_BIND_DN and
-        AUTH_LDAP_BIND_PASSWORD.
+        AUTH_LDAP_BIND_PASSWORD or AUTH_LDAP_BIND_SASL_MECHANISM
+        and AUTH_LDAP_BIND_SASL_CB_VALUE, if defined.
+        """
+        if self.settings.BIND_SASL_MECHANISM:
+            self._bind_sasl(
+                self.settings.BIND_SASL_CB_VALUE,
+                self.settings.BIND_SASL_MECHANISM,
+                sticky=True,
+            )
+        else:
+            self._bind_as(
+                self.settings.BIND_DN, self.settings.BIND_PASSWORD, sticky=True
+            )
+
+    def _bind_sasl(self, cb_value_dict, sasl_mech, sticky=False):
+        """
+        This requires an appropriate cyrus-sasl-* library, ie cyrus-sasl-gssapi
         """
-        self._bind_as(self.settings.BIND_DN, self.settings.BIND_PASSWORD, sticky=True)
+        # <github>/python-ldap/blob/master/Lib/ldap/sasl.py#L41
+        sasl_auth = ldap.sasl.sasl(cb_value_dict, sasl_mech)
+
+        # if the connection has been used for a bind with username/password (to verify)
+        # then we need to clear that before using sasl
+        # <github>/python-ldap/blob/master/Modules/LDAPObject.c#L580
+        # The bind_dn argument will be passed to the c library; howerver,
+        # normally it is not needed and should be an empty string.
+        self._get_connection().sasl_interactive_bind_s("", sasl_auth)
```

Comment:
"non interactive" in this context refers to the sasl method, so that would work for GSSAPI and EXTERNAL it wouldn't work for DIGEST-MD5 and other methods that require other attributes to be set in the auth_cb dict. In python-ldap the non interactive method calls the interactive method with a default auth_cb dict with a blank username. 

Picking which method to call based on the sasl method may be convenient, if the auth_cb dict is empty and one of those two methods is selected - I'll add it.